File: codingTest/programmers/기능 개발.py

def solution(progresses, speeds):
    answer = []

    while progresses:
        progresses = [progresses[i] + speeds[i] for i in range(len(speeds))]
        cnt = 0
        index = 0
        for i in range(len(progresses)):
            if progresses[i] >= 100 :
                cnt += 1
                index = i
            else :
                break

        if cnt > 0 :
            progresses = progresses[index+1 :]
            speeds = speeds[index+1 :]
            answer.append(cnt)

    return answer

if __name__ == "__main__" :
    progresses = [95, 90, 99, 99, 80, 99]
    speeds = [1, 1, 1, 1, 1, 1]
    print(solution(progresses, speeds))

File: codingTestWithSwift.playground/Sources/프로그래머스/기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var result = [Int]()
    
    var time = (0..<progresses.count).reduce(into: [Int](), { target, i in
        var rest = 100 - progresses[i]
        var day = rest / speeds[i]
        if day > 0 && rest % speeds[i] != 0 {
            day += 1
        }
        
        target.append(day)
    })
    
    
    while !time.isEmpty{
        let firstDay = time.removeFirst()
        
        var index = 0
        while index < time.count{
            if firstDay < time[index]{
                break
            }
            index += 1
        }
        
        if index > 0{
            time.removeFirst(index)
            result.append(index+1)
        }else{
            result.append(1)
        }
    }
    
    return result
}
